Primer API Implementation

FlorenzoC opened the pull request:
Dieume què falta i què està malament please

FlorenzoC committed:
Primer API Implementation

Dieume què falta i què està malament please

ifrins commented on backend/app/faceduck/core/signup.py:
```diff
@@ -0,0 +1,14 @@
+
+def validateSignup(user, email, password, name, surname, birthday, gender):
+	if User.get(username = user):
+		return Response({'error-message':'Username used'}, status=400)
```
Jo enlloc de fer returns de `Response` d'algo, el que faria seria tirar excepcions amb l'error (en pots fer una excepció per cada Error message), així es una mica mes type-safe. D'altre banda, també, així no acobles tant la lògica de negoci amb el fet que estiguis fent una API.

ifrins commented on backend/app/faceduck/core/signup.py:
```diff
@@ -0,0 +1,14 @@
+from faceduck import blueprints.py
+from flask import Response
+
+def validateSignup(user, email, password, name, surname, birthday, gender):
```
Aquest mètode te un mal nom perque no només valida, sinó que crea. Jo canviaria el nom

ifrins commented on backend/app/faceduck/core/signup.py:
```diff
@@ -0,0 +1,14 @@
+	newUser = User(username=user, email=email, password=password, name=name, surname=surname, birthday=birthday, gender=gender)
+	newUser.save()
+	
+	return Response(status=204)
```
Ídem com a dalt. Pots retornar el nou user (tot i que despres no en facis res, per exemple).

ifrins commented on backend/app/faceduck/core/signup.py:
```diff
@@ -0,0 +1,14 @@
+from faceduck.models.user import User
+from faceduck import blueprints.py
+from flask import Response
```
Els dos imports últims jo crec que sobren

ifrins commented on backend/app/faceduck/views/api.py:
```diff
@@ -1,5 +1,23 @@
+		birthday = req['birthday']
+		gender = req['gender']
+	except KeyError:
+		resp = {'error-id': '400', 'error-message': 'Invalid data submitted'}
```
Crec que es podria convertir en una exception (o similars) en funció de com ho has fet i així reutilitzes codi (i no dupliques)

3nr1c commented on backend/app/faceduck/core/signup.py:
```diff
@@ -0,0 +1,14 @@
+	if User.get(email=email):
+		return Response({'error-message':'Email used'}, status=400)
+	
+	newUser = User(username=user, email=email, password=password, name=name, surname=surname, birthday=birthday, gender=gender)
```
**S'hauria d'encriptar (hash) la password abans de guardar-la**

FlorenzoC committed:
Canvis varis

Si veieu que està bé podeu fer el merge

FlorenzoC committed:
Error de la cometa

ifrins commented on backend/app/faceduck/core/signup.py:
```diff
@@ -0,0 +1,16 @@
+	except ValueError:
+		return newUser
+	
+	newUser = User(username=user, email=email, password=password, name=name, surname=surname, birthday=birthday, gender=gender)
```
Aquí, a banda, s'ha de setejar el id del usuari (un UUID es bien)

ifrins commented on backend/app/faceduck/core/signup.py:
```diff
@@ -0,0 +1,16 @@
+
+	newUser = None
+	
+	try{
```
no compila 👎

FlorenzoC committed:
Errors canviats

Canvi en el fallo que donava al compilar, implementat uuid, podeu mirar si em compila? M'he barallat un munt d'hores amb docker i no hi ha manera...

ifrins commented on backend/app/faceduck/core/signup.py:
```diff
@@ -0,0 +1,16 @@
+from faceduck.models.user import User
+
+def createUser(id, user, email, password, name, surname, birthday, gender):
```
jo el UUID el crearia dins.

ifrins commented on backend/app/faceduck/core/signup.py:
```diff
@@ -0,0 +1,16 @@
+	except ValueError:
+		return newUser
+	
+	newUser = User(meta='id': id},  username=user, email=email, password=password, name=name, surname=surname, birthday=birthday, gender=gender)
```
![](https://media.giphy.com/media/3o7aD3BJA8vIGVaMw0/giphy.gif)

ifrins replied:
I la PWD continua sense estar encriptada (hash)

ifrins commented on backend/app/faceduck/core/signup.py:
```diff
@@ -0,0 +1,16 @@
+from faceduck.models.user import User
```
I hauries de tornar a fer merge de la branca US-01 a la teva branca per tenir el model

FlorenzoC commented:
Cierro PR ya que he puesto una nueva con el merge de US-01 hecho
